src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -133,6 +133,17 @@
              (query->sql remaining-query entity version paging-options)]
          (jdbc/with-query-results-cursor results-query (comp row-fn munge-fn)))))))
 
+;; Do we still need this, i.e. do we need the pass-through, and the
+;; strict selectivity in the caller below?
```

Comment:
I suppose it could be a lot more efficient to parse 1M top-level vector items one at a time (assuming a very large string), as long as you can avoid holding on to the head.  In any case, at least in the new calls, I just used strict everywhere.